pkg/test/controller/reconciler/testreconciler.go

```diff
@@ -211,67 +264,72 @@ func (r *TestReconciler) NewReconcilerForKind(kind string) reconcile.Reconciler
 	}
 	jg := jitter.NewDefaultGenerator(r.smLoader, dclML)
 
-	switch kind {
-	case "IAMPolicy":
-		reconciler, err = policy.NewReconciler(r.mgr, r.provider, r.smLoader, r.dclConverter, r.dclConfig, immediateReconcileRequests, resourceWatcherRoutines, defaulters, jg)
-	case "IAMPartialPolicy":
-		reconciler, err = partialpolicy.NewReconciler(r.mgr, r.provider, r.smLoader, r.dclConverter, r.dclConfig, immediateReconcileRequests, resourceWatcherRoutines, defaulters, jg)
-	case "IAMPolicyMember":
-		reconciler, err = policymember.NewReconciler(r.mgr, r.provider, r.smLoader, r.dclConverter, r.dclConfig, immediateReconcileRequests, resourceWatcherRoutines, defaulters, jg)
-	case "IAMAuditConfig":
-		reconciler, err = auditconfig.NewReconciler(r.mgr, r.provider, r.smLoader, r.dclConverter, r.dclConfig, immediateReconcileRequests, resourceWatcherRoutines, defaulters, jg)
-	default:
-		crd := testcontroller.GetCRDForKind(r.t, kind)
-		reconciler, err = r.newReconcilerForCRD(crd, defaulters, jg)
+	gvk := u.GroupVersionKind()
+	crd, err := crdloader.GetCRDForGVK(gvk)
+	if err != nil {
+		r.t.Fatal(err)
 	}
+
+	rt, err := ReconcilerTypeForObject(u)
```

Comment:
crdloader.GetCRDForGVK() is called not only in this function, but also when ReconcilerTypeForObject() is called. Can the duplicate API call be avoided?

Reply:
Where is the API call? I felt this was ok, since it seemed like `crdloader.GetCRDForGVK(gvk)` was only reading from disk.

Reply:
You are right that there is no API call. Please find my response at https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2837#discussion_r1791029847 and https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2837#discussion_r1791031633